tests/haskell_proto_library/BUILD

```diff
@@ -0,0 +1,36 @@
+package(default_testonly = 1)
+
+load("@io_tweag_rules_haskell//haskell:haskell.bzl",
+     "haskell_library",
+     "haskell_proto_library",
+)
+
+proto_library(
+  name = "zip_code_proto",
+  srcs = ["zip_code.proto"],
+)
+
+proto_library(
+  name = "address_proto",
+  srcs = ["address.proto"],
+  deps = [":zip_code_proto"],
+)
+
+proto_library(
+  name = "person_proto",
+  srcs = ["person.proto"],
+  deps = [":address_proto"],
+)
+
+haskell_proto_library(
+  name = "person_haskell_proto",
+  deps = [":person_proto"],
+)
+
+haskell_library(
+  name = "hs-lib",
+  srcs = ["Bar.hs"],
+  deps = [":person_haskell_proto"],
+  prebuilt_dependencies = ["base"],
+  visibility = ["//visibility:public"],
+)
```

Comment:
As mentioned, above, I don't think this will work with multiple `haskell_library` rules.  For example:
```
diff --git a/tests/haskell_proto_library/BUILD b/tests/haskell_proto_library/BUILD
index 74efe3a..41526d2 100644
--- a/tests/haskell_proto_library/BUILD
+++ b/tests/haskell_proto_library/BUILD
@@ -10,6 +10,11 @@ proto_library(
   srcs = ["zip_code.proto"],
 )
 
+haskell_proto_library(
+    name = "zip_code_haskell_proto",
+    deps = [":zip_code_proto"],
+)
+
 proto_library(
   name = "address_proto",
   srcs = ["address.proto"],
@@ -30,7 +35,20 @@ haskell_proto_library(
 haskell_library(
   name = "hs-lib",
   srcs = ["Bar.hs"],
-  deps = [":person_haskell_proto"],
+  deps = [
+      ":person_haskell_proto"
+  ],
+  prebuilt_dependencies = ["base"],
+  visibility = ["//visibility:public"],
+)
+
+haskell_library(
+  name = "hs-lib-2",
+  srcs = ["Baz.hs"],
+  deps = [
+      ":person_haskell_proto",
+      ":zip_code_haskell_proto",
+  ],
   prebuilt_dependencies = ["base"],
   visibility = ["//visibility:public"],
 )
```

Example run:
```
$ bazel build :hs-lib-2
INFO: Analysed target //tests/haskell_proto_library:hs-lib-2 (0 packages loaded).
INFO: Found 1 target...
ERROR: /Users/judah.jacobson/repos/rules_haskell/tests/haskell_proto_library/BUILD:45:1: Compiling hs-lib-2 failed (Exit 1)

tests/haskell_proto_library/Baz.hs:4:1: error:
    Ambiguous module name ‘Proto.Tests.HaskellProtoLibrary.ZipCode’:
      it was found in multiple packages:
      testsZShaskellZUprotoZUlibraryZSpersonZUhaskellZUproto-1.0.0 testsZShaskellZUprotoZUlibraryZSzipZUcodeZUhaskellZUproto-1.0.0
  |
4 | import Proto.Tests.HaskellProtoLibrary.ZipCode
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```